src/main/java/teammates/ui/controller/Action.java

```diff
@@ -576,10 +584,131 @@ private boolean isMasqueradeModeRequested(AccountAttributes loggedInUser, String
                && loggedInUser.googleId != null
                && !loggedInUser.googleId.equals(requestedUserId);
     }
+
+    /**
+     *Common method to Get the feedback data
+     */
+
+    protected FeedbackSessionAttributes extractFeedbackSessionData() {
+
+        FeedbackSessionAttributes newSession = new FeedbackSessionAttributes();
+
+        newSession.setCourseId(getRequestParamValue(Const.ParamsNames.COURSE_ID));
+        newSession.setFeedbackSessionName(Sanitizer.sanitizeTitle(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME)));
+
+        // getting the class name from which it's being called
+        String className = new Exception().getStackTrace()[1].getClassName();
+
+        if ("InstructorFeedbackEditSaveAction".equals(className)) {
```

Comment:
@AnaghHegde 
Perhaps this code you're writing belongs in the `InstructorFeedbackEditSaveAction` class itself? Right now you're writing code that only applies to that particular subclass, but you're putting it in the `Action` class, which introduces an unnecessary coupling in the opposite direction (now `Action` needs to know about `InstructorFeedbackEditSaveAction`).

You can override this method (since you've made it `protected`, after all) in the `InstructorFeedbackEditSaveAction` class, after defining a more general method here in the `Action` class.

Also, in the _occasional_ case when you need to actually get the name of the current class, use `this.getClass()`, which returns an object of type `Class<?>` which you can compare directly, or use `this.getClass().getName()` to obtain a string representation. There's no need to construct an exception and get its stack trace; you're misusing the `Exception` object to do something it is not really intended to do, and therefore making your code harder to understand.

Hope this helps. :)